### ISSUE TITLE
Gracefull shutdown (what am I doing wrong?)

### ISSUE THREAD

**Issue by aurelijusrozenas** [php-enqueue/enqueue-dev#964]
Hi,

TLDR; with and without SignalExtension I cannot gracefully shutdown processor by sending any kill signal.

I am using supervisor to run queue processors and supervisor is restarted when app gets deployed. The processors get killed and restarted. So I set out to understand how to gracefully shutdown processor and restart supervisor.
Debugging:
I have processor with `sleep(60)` and I try to gracefully shutdown it but with no luck :/
Supervisor has option to set `stopsignal` e.g. `stopsignal=HUP`. As it did not work now I just trying to `pkill -HUP -f console`. No matter what signal I send processor exits as soon as I send command.

Where do I have a misunderstanding about kill signals or this bundle or what am I doing wrong? I could not find any documentation in this bundle about it except for https://github.com/php-enqueue/enqueue-dev/issues/440, but enabling SignalExtension did not help.

**Comment by gnumoksha** [comment]
might be related to https://github.com/php-enqueue/enqueue-dev/issues/1096
